runtime/astar/src/xcm_config.rs

```diff
@@ -134,6 +141,52 @@ match_types! {
     };
 }
 
+/// A call filter for the XCM Transact instruction. This is a temporary measure until we properly
+/// account for proof size weights.
+///
+/// Calls that are allowed through this filter must:
+/// 1. Have a fixed weight;
+/// 2. Cannot lead to another call being made
+/// 3. Have a defined proof size weight, e.g. no unbounded vecs in call parameters. - TODO: shouldn't max XCM weight handle this?
+pub struct SafeCallFilter;
+
+impl SafeCallFilter {
+    // 1. RuntimeCall::Multisig(..) - contains `Vec` in argument so we should avoid this
+    // 2. RuntimeCall::EVM(..) & RuntimeCall::Ethereum(..) have to be prohibited since we cannot measure PoV size properly
+    // 3. RuntimeCall::Contracts(..) it should be safe to allow for such calls but perhaps it's better to do more delibrate testing on Shibuya/RocStar.
+
+    /// Checks whether the base (non-composite) call is allowed to be executed via `Transact` XCM instruction.
+    pub fn allow_base_call(call: &RuntimeCall) -> bool {
+        match call {
+            RuntimeCall::System(..)
+            | RuntimeCall::Identity(..)
+            | RuntimeCall::Balances(..)
+            | RuntimeCall::Vesting(..)
+            | RuntimeCall::DappsStaking(..)
+            | RuntimeCall::Assets(..)
+            | RuntimeCall::PolkadotXcm(..)
```

Comment:
Not that it affects anything but why do we want to expose _whole_ `pallet_xcm` through Transact, is there an use case where other parachains (as we don't allow DescendOrigin yet) wants to uses `pallet_xcm`'s dispatch?
I see polkadot & kusama runtimes[ only expose `pallet_xcm::limited_reserve_transfer_assets`](https://github.com/paritytech/polkadot/blob/26ff2cc11a6d9f7776c5ed7376f89c37fd214fbe/runtime/kusama/src/xcm_config.rs#L304)

Reply:
I noticed they are more strict in regards to this, however I'm not sure why (yet).

I don't know about what would the scenarios be, but I don't see a problem with them, tbh.
All of the calls are benchmarked (ref time & pov size are known in advance), so it doesn't seem like a vulnerability.

I don't mind limiting this further, but let's try to understand why we do it.